apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/faucet/components/SpinWheel.tsx

```diff
@@ -0,0 +1,176 @@
+"use client";
+
+import { CopyTextButton } from "@/components/ui/CopyTextButton";
+import { Button } from "@/components/ui/button";
+import {
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+import { THIRDWEB_ENGINE_FAUCET_WALLET } from "@/constants/env";
+import { CustomConnectWallet } from "@3rdweb-sdk/react/components/connect-wallet";
+import type { ChainMetadataWithServices } from "app/(dashboard)/(chain)/types/chain";
+import { thirdwebClient } from "lib/thirdweb-client";
+import { useRouter } from "next/navigation";
+import { useState } from "react";
+import Confetti from "react-confetti";
+import { toast } from "sonner";
+import { type ISpinWheelProps, SpinWheel } from "spin-wheel-game";
+import { defineChain } from "thirdweb";
+import { useActiveAccount, useWalletBalance } from "thirdweb/react";
+import { Text } from "tw-components";
+
+const segments = [
+  { segmentText: "Come back tomorrow", segColor: "#f771c9" }, // Mid-dark, optimized for dark mode
+  { segmentText: "0.1", segColor: "#f65ac0" }, // Lightest, optimized for dark mode
+  { segmentText: "Come back tomorrow", segColor: "#f542b7" }, // Slightly darker, optimized
+  { segmentText: "0.01", segColor: "#f32bae" }, // Mid-light, optimized for dark mode
+  { segmentText: "0.05", segColor: "#f213a5" }, // Mid, optimized for dark mode
+  { segmentText: "0.01", segColor: "#da1195" }, // Mid-dark, optimized for dark mode
+  { segmentText: "0.05", segColor: "#c20f84" }, // Darker, optimized for dark mode
+  { segmentText: "0.01", segColor: "#a90d73" }, // Even darker, optimized
+  { segmentText: "0.05", segColor: "#910b63" }, // Near darkest, optimized for dark mode
+  { segmentText: "0.01", segColor: "#790a53" }, // Lightest, optimized for dark mode
+  { segmentText: "0.05", segColor: "#610842" }, // Slightly darker, optimized
+  { segmentText: "0.01", segColor: "#490632" }, // Mid-light, optimized for dark mode
+];
+
+function formatTime(seconds: number) {
+  const hours = Math.floor(seconds / 3600);
+  const minutes = Math.floor((seconds % 3600) / 60);
+
+  if (hours === 0 && minutes <= 1) {
+    return "1 minute";
+  }
+
+  return `${hours}h ${minutes}m`;
+}
+
+export function TestnetSpinWheel({
+  chain,
+  ttlSeconds,
+}: {
+  chain: ChainMetadataWithServices;
+  ttlSeconds: number;
+}) {
+  const address = useActiveAccount()?.address;
+  const chainId = chain.chainId;
+  const definedChain = defineChain(chainId);
+  const evmBalance = useWalletBalance({
+    address: THIRDWEB_ENGINE_FAUCET_WALLET,
+    chain: definedChain,
+    client: thirdwebClient,
+  });
+  const [fundingSent, setFundingSent] = useState(false);
+  const router = useRouter();
+
+  const requestFunds = async (result: string) => {
+    try {
```

Comment:
Prefer `useMutation` with `onSuccess` and `onError` for showing toasts and use mutation status instead of `fundingSet` state

Reply:
I just tried this on a different PR and it didn't work. Is useMutation intended to be used with app router pages?

Reply:
`onSuccess` was being triggered all the time. so handled success and error manually.

Reply:
Looks like @AmineAfia just made this change

Reply:
@arcoraven added this already